src/tests/mocks/erc20_mocks.cairo

```diff
@@ -117,40 +117,40 @@ mod CamelERC20Mock {
         ERC20Event: ERC20Component::Event
     }
 
-    #[constructor]
-    fn constructor(
-        ref self: ContractState,
-        name: felt252,
-        symbol: felt252,
-        initial_supply: u256,
-        recipient: ContractAddress
-    ) {
-        self.erc20.initializer(name, symbol);
-        self.erc20._mint(recipient, initial_supply);
-    }
-
-    #[external(v0)]
-    fn allowance(self: @ContractState, owner: ContractAddress, spender: ContractAddress) -> u256 {
-        self.erc20.allowance(owner, spender)
-    }
-
-    #[external(v0)]
-    fn transfer(ref self: ContractState, recipient: ContractAddress, amount: u256) -> bool {
-        self.erc20.transfer(recipient, amount)
-    }
-
-    #[external(v0)]
-    fn approve(ref self: ContractState, spender: ContractAddress, amount: u256) -> bool {
-        self.erc20.approve(spender, amount)
+    #[abi(per_item)]
```

Comment:
I think we should use `#[abi(embed_v0)]` if everything inside is external. No need for the extra verbosity imo.

Reply:
This trait has the constructor also, but I get what you're saying. Will update

Reply:
*My bad, the reason to use `per_item` is because the compiler gives a warning when trying to embed a non-interface: `Failed to generate ABI: An embedded impl must be an impl of a trait marked with #[starknet::interface].`

We can approach this by using `per_item` in a single trait or we can use loose `external(v0)` functions. I vote for the former because `external(v0)` has been discouraged by SW. WDYT?